File: backend/app/models/fishing_spot.rb
# == Schema Information
#
# Table name: fishing_spots
#
#  id          :uuid             not null, primary key
#  description :text
#  name        :string           not null
#  created_at  :datetime         not null
#  updated_at  :datetime         not null
#
# 釣り場を表すテーブル
class FishingSpot < ApplicationRecord
  has_many :fishing_spot_fishes, dependent: :destroy, inverse_of: :fishing_spot
  has_many :fishes, through: :fishing_spot_fishes
  has_many :locations, class_name: 'FishingSpotLocation', dependent: :destroy, inverse_of: :fishing_spot
  has_many :images, class_name: 'FishingSpotImage', dependent: :destroy, inverse_of: :fishing_spot
end

File: frontend/app/src/interfaces/api/FishingSpot.ts
import { FishingSpotLocation } from './FishingSpotLocation';
import { S3Image } from './s3';
import { Fish } from './Fish';

export interface FishingSpot {
  id: string;
  name: string;
  description: string;
  location: FishingSpotLocation;
  images: S3Image[];
  fishes: Fish[];
}
